src/main/java/duke/parsers/InputParser.java

```diff
@@ -0,0 +1,205 @@
+package duke.parsers;
+
+
+import duke.exceptions.DukeException;
+import duke.filehandler.Storage;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Scanner;
+
+public class InputParser {
+
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+    /***
+     * Checks user inputs, if invalid throws DukeException
+     * @param str User input split by each word
+     * @param task type of task - todo,event,deadline,mark/unmark
+     * @throws DukeException
+     */
+    public static void inputChecker(String[] str, String task) throws DukeException {
```

Comment:
As per our standards, JavaDoc descriptions should [follow the form provided here](https://se-education.org/guides/conventions/java/intermediate.html#comments). Notably, it should only have `/**` at the start, have a line break between description and parameters, and should have punctuation.

```suggestion
    /**
     * Checks user inputs, if invalid throws DukeException.
     *
     * @param str User input split by each word.
     * @param task Type of task, may be one of: "todo", "event", "deadline", "mark/unmark".
     * @throws DukeException If the input is invalid.
     */
```

I notice this also applies to other JavaDocs as well, so do update them appropriately.